### PR TITLE
Feat: Support for Android 12 (SandHook branch)

### PR DESCRIPTION
**Pull request by ghost**
Please create a new branch for those changes.
* Support Android 12
* Remove unused project linking
* Update SandHook to newest edition that support Android 12 ( same as VirtualApp business version )
* Use SlimInlineHook-Lite for inline hooking ( may unstable )
* Kind of another changes......

### CODE REVIEW

**Comment by MlgmXyysd** [comment]
Thank you for your contribution to **true community unity**. We will review it as soon as possible. :)

**Comment by canyie** [comment]
why submit closed-source binaries to a GPL V3 project?

**Comment by google-mirror** [comment]
> why submit closed-source binaries to a GPL V3 project?

why not?

**Comment by Nek0kawa1** [comment]
闭源的 jar 怎么 review 啊，反编译？

**Review by keta1 — APPROVED** [review]

**Review by ojbrc — APPROVED** [review]
.

**Review by mlgm520 — APPROVED** [review]
.

**Comment by google-mirror** [comment]
@MlgmXyysd approved this pull request.

**Comment by reyqn** [comment]
> 
> 
> I decompiled and saw that the closed source file submitted by this PR was fine, and the above questioned the closed source, what is your purpose in posting this? Who instructed you to do this? What is your motive? Did you get permission from the authorities? Did they allow you to post it? Who is behind you and what are you trying to do by posting this? Who are you satirizing? What are you trying to subvert? To undermine what? What are you insinuating?

I'm not sure the way to reassure someone about this is to antagonize them. From what I understand, sandhook was open source before (up to android 11), it seems to be closed source now (for android 12), explaining why those closed source binaries have to be added would probably be more productive (support for android 12 with sandhook can only be made with this "business" closed source edition, right?).

**Comment by MlgmXyysd** [comment]
> > I decompiled and saw that the closed source file submitted by this PR was fine, and the above questioned the closed source, what is your purpose in posting this? Who instructed you to do this? What is your motive? Did you get permission from the authorities? Did they allow you to post it? Who is behind you and what are you trying to do by posting this? Who are you satirizing? What are you trying to subvert? To undermine what? What are you insinuating?
> 
> I'm not sure the way to reassure someone about this is to antagonize them. From what I understand, sandhook was open source before (up to android 11), it seems to be closed source now (for android 12), explaining why those closed source binaries have to be added would probably be more productive (support for android 12 with sandhook can only be made with this "business" closed source edition, right?).

Ignore them, they are just alts from LSPosed for the purpose of attacking us and sending spam to harass. This is what they call "unit the community." I have banned these alts.

**Comment by mglm520** [comment]
> > I decompiled and saw that the closed source file submitted by this PR was fine, and the above questioned the closed source, what is your purpose in posting this? Who instructed you to do this? What is your motive? Did you get permission from the authorities? Did they allow you to post it? Who is behind you and what are you trying to do by posting this? Who are you satirizing? What are you trying to subvert? To undermine what? What are you insinuating?
> 
> I'm not sure the way to reassure someone about this is to antagonize them. From what I understand, sandhook was open source before (up to android 11), it seems to be closed source now (for android 12), explaining why those closed source binaries have to be added would probably be more productive (support for android 12 with sandhook can only be made with this "business" closed source edition, right?).

No. For GPLv3 project to link a closed source business project, it must be granted by the copyright owner and write down the copyright in source files and in the README file. Moreover, purchase voucher should be shown by the purchaser. And only the purchaser can publish this project (but this sometime depends on how the copyright is granted).

Further, in this thread, @MlgmXyysd claimed that we should never use business closed source project:
https://forum.xda-developers.com/t/warning-do-not-use-taichi-any-way.4012681/

So I don't hope this PR be merged. @MlgmXyysd should have the ability to modify the open source version of SandHook.

**Review by ghost — CHANGES_REQUESTED** [review]
.

**Comment by mglm520** [comment]
Why are u deleting my comments? We HAVE the freedom of speech!!

**Review by ghost — CHANGES_REQUESTED** [review]
.

**Comment by liangyuxing520** [comment]
image
Sandhook never has a closed source version.

File: build.gradle [hunk 1]
```diff
@@ -4,7 +4,13 @@ buildscript {
 
     repositories {
         google()
-        jcenter()
+        maven{ url 'https://maven.aliyun.com/nexus/content/repositories/central/'}
+        maven { url 'https://maven.aliyun.com/repository/google'}
+        maven { url 'https://maven.aliyun.com/repository/jcenter'}
```
Comment (qwq233) [inline build.gradle:9]: jcenter已经虚无了吧 加阿里云镜像有有什么必要吗= =

File: build.gradle [hunk 2]
```diff
@@ -44,7 +50,13 @@ allprojects {
     }
     repositories {
         google()
-        jcenter()
+        maven{ url 'https://maven.aliyun.com/nexus/content/repositories/central/'}
+        maven { url 'https://maven.aliyun.com/repository/google'}
+        maven { url 'https://maven.aliyun.com/repository/jcenter'}
```
Comment (qwq233) [inline build.gradle:55]: 同上

**Review by pangsirirat422 — APPROVED** [review]

**Comment by llz572382996** [comment]
大佬，什么时候发布，期待😍

**Pull request closed without merging**